torchgeo/datasets/l7irish.py

```diff
@@ -0,0 +1,293 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+"""L7 Irish dataset."""
+
+import glob
+import os
+import re
+from typing import Any, Callable, Dict, List, Optional, Sequence, cast
+
+import matplotlib.pyplot as plt
+import torch
+from rasterio.crs import CRS
+from torch import Tensor
+
+from .geo import RasterDataset
+from .utils import BoundingBox, download_url, extract_archive
+
+
+class L7Irish(RasterDataset):
+    """L7 Irish dataset.
+
+    The `L7 Irish <https://www.sciencebase.gov/catalog/item/573ccf18e4b0dae0d5e4b109>`__ dataset
+    is based on Landsat 7 Enhanced Thematic Mapper Plus (ETM+) Level-1G scenes.
+    Manually generated cloud masks are used to train and validate cloud cover assessment algorithms,
+    which in turn are intended to compute the percentage of cloud cover in each scene.
+
+    Dataset features:
+
+    * 206 scenes from Landsat-7 ETM+ tiles
+    * Imagery from global tiles between June 2000--December 2001
+    * 9 Level-1 spectral bands with 15 and 30 mper pixel resolution
+
+    Dataset format:
+
+    * Images are composed of multiple single channel geotiffs
+    * Labels are multiclass, stored in a single geotiffs file per image
+    * Level-1 metadata (MTL.txt file)
+    * Landsat-7 ETM+ bands: (B10, B20, B30, B40, B50, B61, B62, B70, B80)
+
+    Dataset classes (4):
+
+    0. Fill
+    1. Clear
+    2. Thin Cloud
+    3. Cloud
+
+    If you use this dataset in your research, please cite the following:
+
+    * https://doi.org/10.5066/F7XD0ZWC
+    * https://doi.org/10.1109/TGRS.2011.2164087
```

Comment:
Can you also add https://www.sciencebase.gov/catalog/item/573ccf18e4b0dae0d5e4b109? That's where most of our data actually comes from